src/localize.cpp

```diff
@@ -376,19 +376,36 @@ void localize_wrk(const BasisSet & basis, arma::mat & C, arma::vec & E, const ar
   }
 }
 
-void localize(const BasisSet & basis, arma::mat & C, arma::vec & E, const arma::mat & P, const arma::mat & H, std::vector<double> occs, bool virt, enum locmet method, enum unitmethod umet, enum unitacc acc, enum startingpoint start, bool delocalize, std::string sizedist, bool size, std::string fname, double Gthr, double Fthr, int maxiter, unsigned long int seed, bool debug) {
+void localize(const BasisSet & basis, arma::mat & C, arma::vec & E, const arma::mat & P, const arma::mat & H, std::vector<double> occs, bool virt, bool core, int numcore, enum locmet method, enum unitmethod umet, enum unitacc acc, enum startingpoint start, bool delocalize, std::string sizedist, bool size, std::string fname, double Gthr, double Fthr, int maxiter, unsigned long int seed, bool debug) {
   // Run localization, occupied space
-  localize_wrk(basis,C,E,P,H,occs,method,umet,acc,start,delocalize,sizedist+".o",size,fname+".o",Gthr,Fthr,maxiter,seed,debug);
+  if (core) {
+    // Run localization, valence space
+    for(size_t i=0;i<numcore;i++)
+      occs[i] = 0.0;
+    localize_wrk(basis,C,E,P,H,occs,method,umet,acc,start,delocalize,sizedist+".o",size,fname+".o",Gthr,Fthr,maxiter,seed,debug);
+  } else {
+    localize_wrk(basis,C,E,P,H,occs,method,umet,acc,start,delocalize,sizedist+".o",size,fname+".o",Gthr,Fthr,maxiter,seed,debug);
+  }
 
   // Run localization, virtual space
   if(virt) {
     for(size_t i=0;i<occs.size();i++)
-      if(occs[i]==0.0)
+      if(occs[i]==0.0 && i >= numcore)
```

Comment:
This is the virtual space, so you shouldn't have any changes here.

Reply:
At this point in the function, if core==true, the occs vector has been left in an altered state. Without the check for i >= numcore, then the core orbitals will be included in the virtual orbital localization (of course, this is not desired!).

I do see an issue, if a user sets numcore to a non-zero value, (even with core==false) this will not work as I expect.

However, this is no longer relevant since I will move to using submatrices as you have suggested.